commit 7563c8e22a0f0abf98538bede9833c023f9306c9
Author: Volker Krause <vkrause@kde.org>
Date:   Mon Sep 16 17:21:29 2024 +0200

    Fix color luma computation for Android window insets
    
    This went unnoticed as it always worked by chance in light mode and we
    didn't have dark mode support yet when this was implemented.

diff --git a/src/android/org/kde/guiaddons/KWindowInsetsController.java b/src/android/org/kde/guiaddons/KWindowInsetsController.java
index 2b8dafb..f8dc316 100644
--- a/src/android/org/kde/guiaddons/KWindowInsetsController.java
+++ b/src/android/org/kde/guiaddons/KWindowInsetsController.java
@@ -25,16 +25,16 @@ public class KWindowInsetsController
 
         if (Build.VERSION.SDK_INT >= 30) {
             if (isDark(color)) {
-                window.getInsetsController().setSystemBarsAppearance(WindowInsetsController.APPEARANCE_LIGHT_STATUS_BARS, WindowInsetsController.APPEARANCE_LIGHT_STATUS_BARS);
-            } else {
                 window.getInsetsController().setSystemBarsAppearance(0, WindowInsetsController.APPEARANCE_LIGHT_STATUS_BARS);
+            } else {
+                window.getInsetsController().setSystemBarsAppearance(WindowInsetsController.APPEARANCE_LIGHT_STATUS_BARS, WindowInsetsController.APPEARANCE_LIGHT_STATUS_BARS);
             }
         } else {
             int visibility = window.getDecorView().getVisibility();
             if (isDark(color)) {
-                visibility |= View.SYSTEM_UI_FLAG_LIGHT_STATUS_BAR;
-            } else {
                 visibility &= ~View.SYSTEM_UI_FLAG_LIGHT_STATUS_BAR;
+            } else {
+                visibility |= View.SYSTEM_UI_FLAG_LIGHT_STATUS_BAR;
             }
             window.getDecorView().setSystemUiVisibility(visibility);
         }
@@ -49,16 +49,16 @@ public class KWindowInsetsController
 
         if (Build.VERSION.SDK_INT >= 30) {
             if (isDark(color)) {
-                window.getInsetsController().setSystemBarsAppearance(WindowInsetsController.APPEARANCE_LIGHT_NAVIGATION_BARS, WindowInsetsController.APPEARANCE_LIGHT_NAVIGATION_BARS);
-            } else {
                 window.getInsetsController().setSystemBarsAppearance(0, WindowInsetsController.APPEARANCE_LIGHT_NAVIGATION_BARS);
+            } else {
+                window.getInsetsController().setSystemBarsAppearance(WindowInsetsController.APPEARANCE_LIGHT_NAVIGATION_BARS, WindowInsetsController.APPEARANCE_LIGHT_NAVIGATION_BARS);
             }
         } else {
             int visibility = window.getDecorView().getVisibility();
             if (isDark(color)) {
-                visibility |= View.SYSTEM_UI_FLAG_LIGHT_NAVIGATION_BAR;
-            } else {
                 visibility &= ~View.SYSTEM_UI_FLAG_LIGHT_NAVIGATION_BAR;
+            } else {
+                visibility |= View.SYSTEM_UI_FLAG_LIGHT_NAVIGATION_BAR;
             }
             window.getDecorView().setSystemUiVisibility(visibility);
         }
@@ -66,7 +66,7 @@ public class KWindowInsetsController
 
     private static double luma(int color)
     {
-        return (0.299 * (color >> 24) + 0.587 * ((color >> 16) | 0xff) + 0.114 * ((color >> 8) | 0xff)) / 255.0;
+        return (0.299 * (color >> 24) + 0.587 * ((color >> 16) & 0xff) + 0.114 * ((color >> 8) & 0xff)) / 255.0;
     }
 
     private static boolean isDark(int color)
